docs/source/design/extract_mapping.rst

```diff
@@ -0,0 +1,492 @@
+
+* ``map_for_vars`` is a dict with keys equal to column names or numeric indices
+  from the source data and values equal to replacement var names.
+
+* ``value_name`` is the melt's resulting value column name.
+
+* ``map_for_values`` is the same as ``value_map``'s ``m``.
+
+.. note::
+
+  Any use of ``melt_map`` can be replaced by multiple uses of ``value_map``
+  with appropriate column stacking.
+
+.. sidebar:: When to use df_map
+
+    When you have no other option.
+
+DF Map
+------
+
+.. code-block:: python
+
+    def df_map(m):
+
+The ``df_map`` operation is for when you just can't figure out how to do what
+you want to do with the other predefined operations. Take a whole DataFrame,
+perform some function ``m`` on it, and return a new DataFrame of the same size.
+
+.. note::
+
+    Needing to use ``df_map`` is a sign that the original data is poorly
+    constructed.
+
+Important Strategic Details
+===========================
+
+Avoid One-off Strings
+---------------------
+
+Avoid using one-off strings for output column headers and values. As much as
+possible, favor variables in ``common.constants`` for cell values and variables
+in ``common.concept_schema.CONCEPT`` for ``out_col`` assignment. That way
+everyone has the chance to use the same shared set of values.
+
+Repeated Information
+--------------------
+
+Multiple identical declarations don't add new information and therefore don't
+matter. This table means exactly the same thing as if it had only one row:
+
+.. csv-table::
+    :header: Participant ID, Specimen ID, Participant Age, Participant Sex, Participant Race
+
+    P1, S1, 7, m, unknown
+    P1, S1, 7, m, unknown
+
+.. _Normalized-Denormalized:
+
+Normalized or Denormalized Doesn't Matter
+-----------------------------------------
+
+Extracting complex files can lead to denormalized output tables. Table
+normalization doesn't matter, because repeated information doesn't matter. If
+creating multiple extraction configurations for a single source data file to
+generate multiple normalized tables that can be manipulated in a Transform
+function is easier for you than making one complex denormalized table for the
+file during Extract, then do it. If not, that's fine too.
+
+.. _Column-Stacking:
+
+Parallel Column Stacking
+------------------------
+
+If the same ``out_col`` is used multiple times, the column output from
+successive operations will be concatenated.
+
+.. image:: /_static/images/columns.svg
+
+Length Rectification
+--------------------
+
+The length disparity between output columns produced by `Parallel Column
+Stacking`_ is then rectified by least-common-multiple repetition of output
+columns which gives a final result that preserves row colinearity from the
+original source data. Note in the next image that A is inline with A, B is
+inline with B, etc.
+
+.. image:: /_static/images/length_rectification.svg
+
+.. caution::
+
+    The lengths of all output columns **must** be multiples of the original
```

Comment:
Maybe add something here that says, if this is not the case an exception is thrown. Just so that the reader knows its something the library checks for you.

Reply:
√ (though that isn't implemented yet)